airflow/models/baseoperator.py

```diff
@@ -742,6 +742,16 @@ def __init__(
                 ]
             )
 
+        if isinstance(self.template_fields, str):
+            warnings.warn(
+                f"The `template_fields` value for {self.task_type} is a string "
+                "but should be a Sequence type. Converting to a Sequence type for execution. "
+                f"Please update {self.task_type} accordingly.",
```

Comment:
```suggestion
                f"The `template_fields` value for {self.task_type} is a string "
                "but should be a list of string. Wrapping it in a list for execution. "
                f"Please update {self.task_type} accordingly.",
```

`str` is _technically_ a string, so _sequence of string_ is a better term. But I feel the term _sequence_ itself might not be the easiest to understand for less experienced developers, so let’s just recommend using a list instead, which is the most fool-proof syntax.

Reply:
Excellent point.